### PR TITLE
Confirmation modal prop types and fix confirm button disabled state logic

### PR DESCRIPTION
**Pull request by stephl3**
## ✍️ Proposed changes

`Modal`
- adds and exports test ids for modal and close button

`ConfirmationModal`
- drops `isRequired` from `buttonText` prop in confirmation modal
- fixes confirm button disabled state logic

🎟 _Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

### For bug fixes, new features & breaking changes

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: .changeset/bright-seas-march.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@leafygreen-ui/confirmation-modal': minor
+---
+
+- Dropped the `isRequired` from the `buttonText` prop type. As of [version 5.1.0](https://github.com/mongodb/leafygreen-ui/blob/main/packages/confirmation-modal/CHANGELOG.md#510), `buttonText` was marked as a deprecated prop that is now optional.
```
Comment (shaneeza) [inline .changeset/bright-seas-march.md:5]: I think that these should all be present tense verbs, so `Drops the ...`, `Fixes existing ...`, and `Adds missing ...`

File: .changeset/bright-seas-march.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+- Fixed existing confirm button disabled state logic. When modal is closed and reopened when interacting with cancel or confirm button, confirm button will conditionally reset to disabled state only when `requiredInputText` is provided.
```
Comment (shaneeza) [inline .changeset/bright-seas-march.md:6]: This sentence is a little hard to follow, `When modal is closed and reopened when interacting with cancel or confirm button, confirm button will conditionally reset to disabled state only when `requiredInputText` is provided.`

Maybe something like, "When the `Modal` is closed using the cancel or confirm button and reopened, the confirm button will ..."

File: .changeset/bright-seas-march.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+- Added missing confirm button disabled state logic. gWhen modal is closed and reopened when interacting with modal close button, confirm button will conditionally reset to disabled state when `requiredInputText` is provided.
```
Comment (shaneeza) [inline .changeset/bright-seas-march.md:7]: Is this repeating the bullet above?

File: packages/confirmation-modal/src/ConfirmationModal/ConfirmationModal.spec.tsx [hunk 4]
```diff
@@ -238,90 +240,90 @@ describe('packages/confirmation-modal', () => {
...
-
-        textInput = getByLabelText('Type "Confirm" to confirm your action');
-
-        expect(textInput).toHaveValue('');
-        expect(confirmationButton).toHaveAttribute('aria-disabled', 'true');
-      });
-    });
+    const requiredInputTextCases = [
+      {
+        describeCase:
```
Comment (shaneeza) [inline packages/confirmation-modal/src/ConfirmationModal/ConfirmationModal.spec.tsx:245]: In this case, we're testing that the confirm button is reset after the users types in the the required test, is that correct? 

We might not need this, but do we test that the disabled button remains disabled after closing the modal if the user does not type in the required text?

Comment (stephl3) [reply packages/confirmation-modal/src/ConfirmationModal/ConfirmationModal.spec.tsx:245]: > In this case, we're testing that the confirm button is reset after the user types in the required text, is that correct?
yes, I left those test cases as-is

> We might not need this, but do we test that the disabled button remains disabled after closing the modal if the user does not type in the required text?
I think the other test blocks that test that the button is enabled by typing in the `requiredInputText` should suffice

File: .changeset/bright-seas-march.md (hunk 3 above)
Comment (stephl3) [reply .changeset/bright-seas-march.md]: updated the language for clarity. 2nd bullet addresses fixing cancel/confirm button while the 3rd bullet addresses the missing logic for the modal close button ([this line](https://github.com/mongodb/leafygreen-ui/pull/2367/files#diff-28b99ffb5f3e5a00a3ae8ac9b42c39632b5973489941fc206fade38025a31dabL97-R102))

File: .changeset/bright-seas-march.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+---
+'@leafygreen-ui/confirmation-modal': minor
+---
+
+- Drops the `isRequired` from the `buttonText` prop type. As of [version 5.1.0](https://github.com/mongodb/leafygreen-ui/blob/main/packages/confirmation-modal/CHANGELOG.md#510), `buttonText` is a deprecated prop that is now optional.
+- Fixes existing confirm button disabled state logic
+  - Previously, when the modal was closed using the cancel or confirm button and reopened, the confirm button would always reset to a disabled state although it should not reset when `requiredInputText` is provided.
+  - Now, the confirm button will reset to a disabled state when `requiredInputText` is provided.
```
Comment (shaneeza) [inline .changeset/bright-seas-march.md:8]: Do these two bullets contradict each other? 

> the confirm button would always reset to a disabled state although it should not reset when `requiredInputText` is provided

and 

> Now, the confirm button will reset to a disabled state when `requiredInputText` is provided.

Comment (stephl3) [reply .changeset/bright-seas-march.md]: good catch, updated! I think it should read better now 😅

File: .changeset/bright-seas-march.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+---
+'@leafygreen-ui/confirmation-modal': minor
+---
+
+- Drops the `isRequired` from the `buttonText` prop type. As of [version 5.1.0](https://github.com/mongodb/leafygreen-ui/blob/main/packages/confirmation-modal/CHANGELOG.md#510), `buttonText` is a deprecated prop that is now optional.
+- Fixes existing confirm button disabled state logic
+  - Previously, when the modal was closed using the cancel or confirm button and reopened, the confirm button would always reset to a disabled state disregarding `requiredInputText` prop value.
```
Comment (shaneeza) [inline .changeset/bright-seas-march.md:7]: you're missing a comma here: `disabled state, disregarding`

File: .changeset/bright-seas-march.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+  - Now, the confirm button will reset to a disabled state when `requiredInputText` is provided and and not reset to a disabled state when `requiredInputText` is undefined.
```
Comment (shaneeza) [inline .changeset/bright-seas-march.md:8]: and ~and~ not

Comment (shaneeza) [reply .changeset/bright-seas-march.md:8]: actually, you might not even need this bit: 

> and not reset to a disabled state when `requiredInputText` is undefined.

The first sentence makes it clear that it will only reset when `requiredInputText` is provided

File: .changeset/bright-seas-march.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+- Adds missing confirm button disabled state logic
+  - Previously, when the modal was closed using the modal close button and reopened, the confirm button would never reset to a disabled state.
+  - Now, the confirm button will reset to a disabled state when `requiredInputText` is provided and not reset to a disabled state when `requiredInputText` is undefined.
```
Comment (shaneeza) [inline .changeset/bright-seas-march.md:11]: Same as above, you might not need:

> and not reset to a disabled state when `requiredInputText` is undefined.

**Review by shaneeza — APPROVED** [review]
Looks good. I added a few nits.
